src/stateManagement/imageIdSpecificStateManager.js

```diff
@@ -35,17 +35,19 @@ function newImageIdSpecificToolStateManager() {
   // Here we add tool state, this is done by tools as well
   // As modules that restore saved state
   function addImageIdSpecificToolState(element, toolType, data) {
-    const enabledImage = external.cornerstone.getEnabledElement(element);
-    // If we don't have any tool state for this imageId, add an empty object
+    const enabledElement = external.cornerstone.getEnabledElement(element);
 
-    if (
-      !enabledImage.image ||
-      toolState.hasOwnProperty(enabledImage.image.imageId) === false
```

Comment:
Yes ... but the error occurs a few lines later on the previous line 45 (moved in my commit to line 47).
```js
toolState[enabledImage.image.imageId] = {};
```

If `enabledElement.image` is undefined, line 45 will throw because you're trying to access a property of undefined.

This is why the other methods are not affected by this problem. Within their `if` clauses, they all `return`.